### ISSUE TITLE
Curly-braces indentation bug in editor

### ISSUE THREAD

**Issue by myisaak** [ruby/TryRuby#36]
There is a particular annoyance when going through an interactive web tutorial which lies in the auto-indenting of your code -- and by auto-indenting I mean to avoid pressing TAB.

In your interactive Ruby tutorial, and playground [this was the case](https://ruby.github.io/TryRuby/playground/). To reproduce this bug, take for example the following uncompleted ruby playground environment, note the use of `->` to mark indents.
```
def main
->5.times {|i|
->->puts i
-># ...
end
```
When I add the second curly brace `}` after the third line `puts i`, what I expect is one indent like `->}` but what I get is the following
```
def main
->5.times {|i|
->->puts i
->->} # unnecessary indent
end
```
There is an extra indent, or in other words, a missing or faulty auto formatting.
How exactly did you guys implement your indentation, I wouldn't mind submitting a pull request.

**Comment by easydatawarehousing** [comment]
Hi Isaak,

Thanks for reporting this issue. I've looked into this a little bit. The editor window uses a javascript library called [codemirror](https://codemirror.net/). The minified and bundled version can be found in src/javascripts/vendor/codemirror-compressed-4.8.js. This version is quite old, current version is 5.42.
However the `}` de-indentation problem still seems to be present in the latest version (clone the codemirror repo, open mode/ruby/index.html). The keyword `end` is de-indented but `}` isn't.

Personally I'm not very eager to dive into codemirror code and fix this. But don't let that stop you :) I'm happy to update the vendored codemirror library.

Thanks,
Ivo

**Comment by myisaak** [comment]
A pull request for `codemirror` that I created fixed this issue codemirror/CodeMirror#5881 in version 5.51.0.

An update will be needed on codemirror from the current https://github.com/ruby/TryRuby/blob/84ec432786e9f38dc27477f974583d6a7710be30/source/javascripts/vendor/codemirror-compressed-4.8.js#L1  to at least 5.51.0 to close this issue.
